### PR TITLE
Explicitly allow static builds to be disabled and add support for non standard zlib and bzip2 in Amber easyblock

### PR DESCRIPTION
**Pull request by valtandor**
…standard zlib and bzip2

### CODE REVIEW

File: easybuild/easyblocks/a/amber.py [hunk 1]
```diff
@@ -141,9 +143,17 @@ def install_step(self):
                 self.mpi_option = '-mpi'
 
         common_configopts = [self.cfg['configopts'], '--no-updates', '-noX11']
-        if self.name == 'Amber':
+        if self.name == 'Amber' and self.cfg['static']:
             common_configopts.append('-static')
 
+        envval = lambda ev: os.getenv(ev) or ''
```
Comment (boegel) [inline easybuild/easyblocks/a/amber.py]: @valtandor You don't need this, you can use `os.getenv('AMBERBUILDFLAGS', '')` instead

File: easybuild/easyblocks/a/amber.py [hunk 2]
```diff
@@ -141,9 +143,17 @@ def install_step(self):
...
+        zlibroot = get_software_root('zlib')
+        if zlibroot:
+            env.setvar('AMBERBUILDFLAGS', envval('AMBERBUILDFLAGS') + " -L" + os.path.join(zlibroot, "lib"))
+        bzip2root = get_software_root('bzip2')
+        if bzip2root:
+            env.setvar('AMBERBUILDFLAGS', envval('AMBERBUILDFLAGS') + " -L" + os.path.join(bzip2root, "lib"))
```
Comment (boegel) [inline easybuild/easyblocks/a/amber.py]: You can simplify this a little bit by making it a loop:

```python
for dep in ['bzip2', 'zlib']:
    root = get_software_root(dep)
    amberbuildflags = os.getenv('AMBERBUILDFLAGS', '')
    env.setvar('AMBERBUILDFLAGS', amberbuildflags + " -L" + os.path.join(root, 'lib'))

**Comment by akesandgren** [comment]
The "static" flag functionality has been merged in PR #1468 (copied from this PR)
There should be no need to touch AMBERBUILDFLAGS, bzip2 and zlib will be picked up from LD_LIBRARY_PATH and when having the Amber module loaded cpptraj and ambpdb will use the correct search path.

**Comment by boegel** [comment]
This has bitrotted, and it seems like the relevant stuff has been merged already via #1468, so closing...

**Pull request closed without merging**